app/ante/min_fee_test.go

```diff
@@ -0,0 +1,118 @@
+package ante_test
+
+import (
+	"math"
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/ante"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	version "github.com/tendermint/tendermint/proto/tendermint/version"
+)
+
+func TestCheckTxFeeWithGlobalMinGasPrices(t *testing.T) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+
+	builder := encCfg.TxConfig.NewTxBuilder()
+	err := builder.SetMsgs(banktypes.NewMsgSend(
+		testnode.RandomAddress().(sdk.AccAddress),
+		testnode.RandomAddress().(sdk.AccAddress),
+		sdk.NewCoins(sdk.NewInt64Coin(appconsts.BondDenom, 10))),
+	)
+	require.NoError(t, err)
+
+	feeAmount := int64(1000)
+
+	ctx := sdk.Context{}
+
+	testCases := []struct {
+		name       string
+		fee        sdk.Coins
+		gasLimit   uint64
+		appVersion uint64
+		expErr     bool
+	}{
+		{
+			name:       "bad tx; fee below required minimum",
+			fee:        sdk.NewCoins(sdk.NewInt64Coin(appconsts.BondDenom, feeAmount-1)),
+			gasLimit:   uint64(float64(feeAmount) / appconsts.GlobalMinGasPrice),
+			appVersion: uint64(2),
+			expErr:     true,
+		},
+		{
+			name:       "good tx; fee equal to required minimum",
+			fee:        sdk.NewCoins(sdk.NewInt64Coin(appconsts.BondDenom, feeAmount)),
+			gasLimit:   uint64(float64(feeAmount) / appconsts.GlobalMinGasPrice),
+			appVersion: uint64(2),
+			expErr:     false,
+		},
+		{
+			name:       "good tx; fee above required minimum",
+			fee:        sdk.NewCoins(sdk.NewInt64Coin(appconsts.BondDenom, feeAmount+1)),
+			gasLimit:   uint64(float64(feeAmount) / appconsts.GlobalMinGasPrice),
+			appVersion: uint64(2),
+			expErr:     false,
+		},
+		{
+			name:       "good tx; with no fee (v1)",
+			fee:        sdk.NewCoins(sdk.NewInt64Coin(appconsts.BondDenom, feeAmount)),
+			gasLimit:   uint64(float64(feeAmount) / appconsts.GlobalMinGasPrice),
+			appVersion: uint64(1),
+			expErr:     false,
+		},
+		{
+			name:       "good tx; gas limit and fee are maximum values",
+			fee:        sdk.NewCoins(sdk.NewInt64Coin(appconsts.BondDenom, math.MaxInt64)),
+			gasLimit:   math.MaxUint64,
+			appVersion: uint64(2),
+			expErr:     false,
+		},
+		{
+			name:       "bad tx; gas limit and fee are 0",
+			fee:        sdk.NewCoins(sdk.NewInt64Coin(appconsts.BondDenom, 0)),
+			gasLimit:   0,
+			appVersion: uint64(2),
+			expErr:     true,
+		},
+		{
+			name:       "good tx; rounds down to 0 gets reset to 1",
```

Comment:
[nit] it's not immediately obvious what gets rounded down to 0 and reset to 1.

Perhaps a better test name would be "expect no error when minFee = 0.8"

Related, should there be another test case like:

```go
		{
			name:       "expect error when tx fee is 0",
			fee:        sdk.NewCoins(sdk.NewInt64Coin(appconsts.BondDenom, 0)),
			gasLimit:   400,
			appVersion: uint64(2),
			expErr:     false,
		},
```

Reply:
we're allowing tx fee of 0